pkg/common/utils/utils.go

```diff
@@ -95,3 +95,40 @@ func QueryAllVolumeUtil(ctx context.Context, m cnsvolume.Manager, queryFilter cn
 	}
 	return queryResult, nil
 }
+
+func QuerySnapshotsUtil(ctx context.Context, m cnsvolume.Manager, snapshotQueryFilter cnstypes.CnsSnapshotQueryFilter) ([]cnstypes.CnsSnapshotQueryResultEntry, error) {
+	log := logger.GetLogger(ctx)
+	var allQuerySnapshotResults []cnstypes.CnsSnapshotQueryResultEntry
+	var snapshotQuerySpec cnstypes.CnsSnapshotQuerySpec
+	if snapshotQueryFilter.SnapshotQuerySpecs == nil {
+		log.Infof("Attempting to retrieve all the Snapshots available in the vCenter inventory.")
+	} else {
+		snapshotQuerySpec = snapshotQueryFilter.SnapshotQuerySpecs[0]
+		log.Infof("Invoking QuerySnapshots with spec: %+v", snapshotQuerySpec)
+	}
+	iteration := 1
+	for {
+		snapshotQueryResult, err := m.QuerySnapshots(ctx, snapshotQueryFilter)
+		if err != nil {
+			return nil, logger.LogNewErrorCodef(log, codes.Internal,
+				"querySnapshots failed for snapshotQueryFilter: %v. Err=%+v", snapshotQueryFilter, err.Error())
+		}
+		if snapshotQueryResult == nil {
+			log.Infof("Observed empty SnapshotQueryResult")
+			break
+		}
+		if len(snapshotQueryResult.Entries) == 0 {
+			log.Infof("QuerySnapshots retrieved no results for the spec: %+v", snapshotQuerySpec)
+		}
+		allQuerySnapshotResults = append(allQuerySnapshotResults, snapshotQueryResult.Entries...)
+		log.Infof("%v more snapshots to be queried", snapshotQueryResult.Cursor.TotalRecords-snapshotQueryResult.Cursor.Offset)
+		if snapshotQueryResult.Cursor.Offset == snapshotQueryResult.Cursor.TotalRecords {
+			log.Infof("QuerySnapshots retrieved all records (%d) for the SnapshotQuerySpec: %+v in %d iterations",
+				snapshotQueryResult.Cursor.TotalRecords, snapshotQuerySpec, iteration)
+			break
+		}
+		iteration++
```

Comment:
Have you tested this when iteration > 1?

Reply:
Yes,
```
{"level":"info","time":"2021-07-12T13:15:24.286366-07:00","caller":"utils/utils.go:126","msg":"QuerySnapshots retrieved all records (2) for the SnapshotQuerySpec: {DynamicData:{} VolumeId:{DynamicData:{} Id:} SnapshotId:<nil>} in 2 iterations"}
    utils_test.go:154: Snapshots:
    utils_test.go:156: {{} {{} {{} 758d4144-67e6-4058-a497-81e1e1a4aec6} {{} 8d2e4b62-d073-4eb0-b84d-ade17ac0d6bf} cns1 2021-07-07 22:34:38.079999 +0000 UTC} <nil>}
    utils_test.go:156: {{} {{} {{} 2ca22363-4698-47e4-9ac9-5cbb2f2b517c} {{} 8d2e4b62-d073-4eb0-b84d-ade17ac0d6bf} cns2 2021-07-07 22:35:03.81 +0000 UTC} <nil>}
--- PASS: TestQuerySnapshotsUtil (0.51s)
```